mods/xpanes/init.lua

```diff
@@ -1,156 +1,146 @@
-xpanes = {}
 
-local function rshift(x, by)
-	return math.floor(x / 2 ^ by)
+local function is_pane(pos)
+	return minetest.get_item_group(minetest.get_node(pos).name, "pane") > 0
 end
 
-local directions = {
-	{x = 1, y = 0, z = 0},
-	{x = 0, y = 0, z = 1},
-	{x = -1, y = 0, z = 0},
-	{x = 0, y = 0, z = -1},
-}
-
-local function update_pane(pos, name)
-	if not minetest.get_node(pos).name:find("^xpanes:"..name) then
-		return
+local function connects_dir(pos, name, dir)
+	local aside = vector.add(pos, minetest.facedir_to_dir(dir))
+	if is_pane(aside) then
+		return true
 	end
-	local sum = 0
-	for i, dir in pairs(directions) do
-		local node = minetest.get_node(vector.add(pos, dir))
-		local def = minetest.registered_nodes[node.name]
-		local pane_num = def and def.groups.pane or 0
-		if pane_num > 0 or not def or (def.walkable ~= false and
-				def.drawtype ~= "nodebox") then
-			sum = sum + 2 ^ (i - 1)
-		end
+
+	local connects_to = minetest.registered_nodes[name].connects_to
+	if not connects_to then
+		return false
 	end
-	if sum == 0 then
-		sum = 15
+	local list = minetest.find_nodes_in_area(aside, aside, connects_to)
+
+	if #list > 0 then
+		return true
 	end
-	minetest.set_node(pos, {name = "xpanes:"..name.."_"..sum})
+
+	return false
 end
 
-local function update_nearby(pos, node)
-	node = node or minetest.get_node(pos)
-	local name = node.name
-	if not name or node.name:sub(1, 7) ~= "xpanes:" then
+local function swap(pos, node, name, param2)
```

Comment:
this seems confusingly names to me, it's more of a set_node which avoids excess work